src/scripts/modules/media/body/body.coffee

```diff
@@ -125,12 +134,201 @@ define (require) ->
           $temp.find('ol[start], [data-type="list"][data-list-type="enumerated"][start]').each (i, el) ->
             $el = $(el)
             $el.css('counter-reset', 'list-item ' + $el.attr('start'))
+
+          # # uncomment to embed fake exercises and see embeddable exercises in action
+          # @fakeExercises($temp)
+
+          @initializeEmbeddableQueues()
+          @findEmbeddables($temp.find('#content'))
+
       catch error
         # FIX: Log the error
         console.log error
 
       @$el?.html($temp.html())
 
+
+
+    ###
+    # For handling embeddables (exercises from Tutor, concept coaches, simulations, etc).
+    # Could abstract a little more and pull out to another file if needed.
+    ###
+
+    # A queue for rendering embeddables means
+    # we can share rendering functions regardless
+    # of whether the embeddable is sync or async
+    initializeEmbeddableQueues: () ->
+      @renderEmbeddableQueue = []
+
+      _.extend(@renderEmbeddableQueue, Backbone.Events)
+
+      @renderEmbeddableQueue?.on('add', @renderEmbeddable)
+      # Uncomment to add action for messages
+      #
+      # There's only one right now for when all the asyncs are done
+      # @renderEmbeddableQueue?.on('message', (message)->)
+      #
+      # Post render again when all promises have returned
+      @renderEmbeddableQueue?.on('done', @onRender.bind(@))
+
+    renderEmbeddable : (embeddableItem) =>
+      # finds fresh element in @$el if a selector is provided
+      # instead of the element itself.
+
+      embeddableItem.$el = @$el.find(embeddableItem.selector) if embeddableItem.selector
+
+      $parent = embeddableItem.$el.parent()
+      embeddableItem.$el.replaceWith(embeddableItem.html)
+
+      if _.isFunction(embeddableItem.onRender)
+        embeddableItem.onRender($parent)
```

Comment:
Coffeescript offers an easier way to check if something is a function:
```coffeescript
embeddableItem.onRender?($parent)
```

Compiles to:
```
if (typeof embeddableItem.onRender === "function") {
  embeddableItem.onRender($parent);
}
```